pkg/controller/appdefinition/pullappimage.go

```diff
@@ -11,9 +12,38 @@ import (
 	"github.com/acorn-io/acorn/pkg/tags"
 	"github.com/acorn-io/baaah/pkg/router"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type imageClient interface {
+	remoteImageDigest(ctx context.Context, c client.Reader, namespace, image string) (string, error)
+	imagePullAppImage(ctx context.Context, c client.Reader, namespace, image, nestedDigest string) (*v1.AppImage, error)
```

Comment:
Curious why this starts with "image" instead of just being called "pullAppImage"

Reply:
this function calls a different function. unfortunately the name were very similar to begin with which i thought created confusion.
imagePullAppImage was a way to remember this one directly calls image package PullAppImage function which is different from the main PullAppImage function in question here.